kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+package org.kitodo.mediaserver.importer.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.filefilter.FalseFileFilter;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.kitodo.mediaserver.importer.config.ImporterProperties;
+import org.kitodo.mediaserver.importer.exceptions.ImporterException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Utilities for the importer.
+ */
+public class ImporterUtils {
+
+    private ImporterProperties importerProperties;
+
+    @Autowired
+    public void setImporterProperties(ImporterProperties importerProperties) {
+        this.importerProperties = importerProperties;
+    }
+
+    /**
+     * Searches the import hotfolder for a new work to import. If a work is found, it is moved to a subdirectory of
+     * the import-in-progress folder.
+     *
+     * <p>
+     * Checks first if there are xml files directly in the hotfolder. If so, it is assumed that this is a METS/MODS
+     * file, and this file and all directories with the xml file name as prefix are moved into a new subfolder in the
+     * importing directory.
+     *
+     * <p>
+     * If no xml file is found, the first directory found is copied to the importing directory, assuming this is a
+     * process folder for the complete work.
+     *
+     * @return the directory with the work to import, null if no work is found.
+     * @throws Exception if a severe error occurs.
+     */
+    public File getWorkPackage() throws Exception {
+        File hotfolder = new File(importerProperties.getHotfolderPath());
+        File importingFolder  = new File(importerProperties.getImportingFolderPath());
+        importingFolder.mkdirs();
+
+        if (hotfolder == null || !hotfolder.isDirectory()) {
+            throw new ImporterException("The hotfolder '" + importerProperties.getHotfolderPath()
+                    + "' is not a directory, please check your configuration");
+        }
+        if (importingFolder == null) {
+            throw new ImporterException("The import-in-progress folder '" + importerProperties.getImportingFolderPath()
+                    + "' is not defined, please check your configuration");
+        }
+        importingFolder.mkdirs();
```

Comment:
This is already called on line 59. Is this supposed to be so?

Reply:
removed line 59